webapp/src/components/ImageFieldEditor.tsx

```diff
@@ -186,6 +186,24 @@ export class ImageFieldEditor<U extends pxt.Asset> extends React.Component<Image
             assets = assets.map(t => (t.type !== this.asset.type || t.id !== this.asset.id) ? t : assetToGalleryItem(this.getValue()))
         }
 
+        if (filterString) {
+            assets.forEach(a => {
+                if (!a.meta.tags && this.options) {
+                    a.meta.tags = this.blocksInfo.apis.byQName[a.id]?.attributes.tags?.split(" ") || [""];
+                }})
+
+            const tags = filterString.split(" ")
```

Comment:
It looks like the pxt.sprite.filter version is used in a few spots and would be a bit tedious to dedup due to the different interfaces so probably fine, but if we're duplicating this would be good to add a comment in both spots pointing at eachother to avoid them getting out of sync